components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link_close_button.jsx

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+import PropTypes from 'prop-types';
+
+import {Constants} from 'utils/constants.jsx';
+
+export default class SidebarChannelButtonOrLinkCloseButton extends React.PureComponent {
+    static propTypes = {
+        handleClose: PropTypes.func,
+        channelId: PropTypes.string.isRequired,
+        channelType: PropTypes.string.isRequired,
+        teammateId: PropTypes.string,
+        badge: PropTypes.string
+    }
+
+    overlayTriggerAttr = ['hover', 'focus']
+
+    handleClose = (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+        this.props.handleClose();
+    }
+
+    render() {
+        let closeButton = null;
+
+        let removeTooltip = (
```

Comment:
Could you move the code to set the tooltip into the `if (this.props.handleClose && !this.props.badge)` block to make it clear that that's only used when we have a close handler? I thought there was some new functionality here that I had missed

Reply:
Done